cmd/init.go

```diff
@@ -59,57 +66,62 @@ func Init() *cobra.Command {
 				return err
 			}
 
-			if err := executeInit(devPath, overwrite, l, workDir); err != nil {
+			if err := executeInit(namespace, devPath, l, workDir, overwrite); err != nil {
 				return err
 			}
 
-			log.Success(fmt.Sprintf("Okteto manifest (%s) created", devPath))
+			log.Success(fmt.Sprintf("Okteto manifest (%s) created. Check its content in case you need to adapt it", devPath))
+			log.Information("Run 'okteto up' to activate your development container")
 			return nil
 		},
 	}
 
+	cmd.Flags().StringVarP(&namespace, "namespace", "n", "", "namespace target for generating the okteto manifest")
 	cmd.Flags().StringVarP(&devPath, "file", "f", defaultManifest, "path to the manifest file")
 	cmd.Flags().BoolVarP(&overwrite, "overwrite", "o", false, "overwrite existing manifest file")
 	return cmd
 }
 
-func executeInit(devPath string, overwrite bool, language string, workDir string) error {
-	if !overwrite {
-		if model.FileExists(devPath) {
-			return fmt.Errorf("%s already exists. Please delete it before running the command again", devPath)
-		}
+func executeInit(namespace, devPath, language, workDir string, overwrite bool) error {
+	devPath, err := validateDevPath(devPath, overwrite)
+	if err != nil {
+		return err
 	}
 
-	var dev *model.Dev
-	if len(language) > 0 {
-		log.Debugf("generating config for %s", language)
-		dev = linguist.GetDevConfig(language)
-	} else {
-		l, err := linguist.ProcessDirectory(workDir)
-		if err != nil {
-			log.Info(err)
-			return fmt.Errorf("Failed to determine the language of the current directory")
-		}
+	ifAskingForDeployment := language == ""
 
-		language = l
-		if language == linguist.Unrecognized {
-			l, err := askForLanguage()
-			if err != nil {
-				return err
-			}
-
-			language = l
-		}
-
-		dev = linguist.GetDevConfig(language)
-		dev.Image = askForImage(language, dev.Image)
+	language, err = getLanguage(language, workDir)
+	if err != nil {
+		return err
 	}
-	var err error
-	dev.Name, err = model.GetValidNameFromFolder(workDir)
+
+	dev, err := linguist.GetDevDefaults(language, workDir, ifAskingForDeployment)
 	if err != nil {
 		return err
 	}
 
+	if ifAskingForDeployment {
+		d, container, err := getDeployment(namespace)
+		if err != nil {
+			return err
+		}
+		dev.Container = container
+		if container == "" {
+			container = d.Spec.Template.Spec.Containers[0].Name
+		}
+
+		postfix := fmt.Sprintf("Analyzing deployment '%s'...", d.Name)
```

Comment:
put the spinner before getting the deployment to it's also shown while doing the get